### DISCUSSION TITLE
Plugin for DynamoDB

### DISCUSSION THREAD

**Discussion by taleena** [grafana/grafana#51750] [Data Sources]
As a Grafana user, I'd like a plugin for DynamoDB: https://aws.amazon.com/dynamodb/

**Comment by fernandopiacente** [comment]
That would be awesome!

**Comment by ravikumargrk** [comment]
I need this too !

**Comment by obue** [comment]
That would solve on our side also a lot of problems.

**Reply by VermaPriyanka** [reply]
@Goldich Can you share some use-cases that you would like to address with the DynamoDB plugin?

**Reply by obue** [reply]
@VermaPriyanka Sure I can explain. 

Use-Case 1: Visualization of Semi Structured data from DynamoDB
We use a lot of tables in our solution to store configuration data. These data sets are helpful as context information for existing dashboards to add meta data. At the moment we need to add all data in the data ingest pipeline to have them available in dashboards. I would like to argument them to generate more insights. 

Use-Case 2: Templates
I have a set of Standardized Dashboards deployed via Terraform for multiple customers. I would like to make parts of the dashboard dynamic and code independent to change it at runtime. The problem is, that some information is not available at initial provisioning and the effort of maintaining it in Terraform is a too high
 operations effort.

Use-Case 3: View of a DynamoDB table
As simple as it can be. View the content of a table for users outside AWS. DynamoDB has no resource policy's and it's easier to control the permissions in Grafana. 

I hope that helps you 😉

**Comment by alvaro-ponce** [comment]
+1

**Comment by Hatko** [comment]
+1

**Comment by nafernandafw** [comment]
+1

**Comment by syadnom** [comment]
+1

**Comment by bidetzz** [comment]
+1

**Comment by ouno-home** [comment]
+1

**Comment by shim517** [comment]
I need this too!
+1

**Comment by TheOnlyDavid** [comment]
+1

**Comment by FlanniganColum** [comment]
Also a requirement and would just help to enrich Grafana
+1

**Comment by musiczhzhao** [comment]
+1 

What's the current status of this feature?

**Comment by Rusty-ScoutII** [comment]
+1

**Comment by yosiasz** [comment]
> As a Grafana user, I'd like a plugin for DynamoDB: https://aws.amazon.com/dynamodb/

does dynamodb provide a rest api endpoint? can that be leveraged

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I've re-opened the issue at #16265 and will close this discussion.

**Comment by scottlepp** [comment]
We've released a [beta version](https://grafana.com/grafana/plugins/grafana-dynamodb-datasource/).  It currently supports [PartiQL](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ql-gettingstarted.html).
